### DISCUSSION TITLE
set getRelations based on user roles

### DISCUSSION THREAD

**Discussion by kcanyildirim** [filamentphp/filament#6929] [Help]
Hello, how can I set the relationships based on user roles? For example CoursesRelationManager, UserNotesRelationManager if Student or if teacher only show CoursesRelationManager, UserBranchesRelationManager.

` return [
            CoursesRelationManager::class,
            UserNotesRelationManager::class,
            UserBranchesRelationManager::class,
        ];
`

**Comment by pboivin** [comment]
I think relation managers will follow your Laravel policies, but maybe I'm missing some detail... Have you tried something like this?
```php
    $relations = [];

    if (CourseResource::canViewAny()) {
        $relations[] = CoursesRelationManager::class;
    }

    // ...

    return $relations;
```

**Reply by kcanyildirim** [reply]
Actually I want to use this in user edit page. I need to do it by record id not auth. I tried a code like this but I get an error.

```
    public static function getRelations(): array
    {
        $relations = [];
        $record_id = request()->route('record');
        if($record_id) {
            $user = User::find($record_id);

            if ($user->hasRole('Teacher')) {
                $relations[] = CoursesRelationManager::class;
                $relations[] = UserNotesRelationManager::class;
                $relations[] = UserBranchesRelationManager::class;
            } elseif ($user->hasRole('Student')) {
                $relations[] = CoursesRelationManager::class;
                $relations[] = UserNotesRelationManager::class;
            }
        }

        return $relations;
    }
```

**Reply by kcanyildirim** [reply]
Unable to find component: [app.filament.resources.user-course-resource.relation-managers.courses-relation-manager]

**Reply by pboivin** [reply]
You can't rely on `request()` in the static methods of the resource. I think you'll need to use `getRelationManagers()` on the Page class, but I don't really have a lot of experience customizing relation managers...

**Comment by mginod** [comment]
I wanted to achieve the same thing in a Resource shared between different panels. My solution was to override the method `canViewForRecord` of a `RelationManager`:

```
// In my RelationManager
public static function canViewForRecord(Model $ownerRecord, string $pageClass): bool
{
    return auth()->user()->isAdmin();
}
```
